backend/LexBoxApi/GraphQL/ProjectMutations.cs

```diff
@@ -311,6 +311,28 @@ await dbContext.ProjectUsers
         return dbContext.Projects.Where(p => p.Id == input.ProjectId);
     }
 
+    [Error<NotFoundException>]
+    [Error<DbError>]
+    [UseMutationConvention]
+    [UseFirstOrDefault]
+    [UseProjection]
+    public async Task<IQueryable<Project>> SetRetentionPolicy(
+        SetRetentionPolicyInput input,
+        IPermissionService permissionService,
+        [Service] ProjectService projectService,
+        LexBoxDbContext dbContext)
+    {
+        await permissionService.AssertCanManageProject(input.ProjectId);
+        var project = await dbContext.Projects.FindAsync(input.ProjectId);
+        NotFoundException.ThrowIfNull(project);
+
+        project.RetentionPolicy = input.RetentionPolicy;
+        project.UpdateUpdatedDate();
+        projectService.InvalidateProjectConfidentialityCache(input.ProjectId);
```

Comment:
The project confidentiality cache is not affected by the retention policy changing; this line can be removed.